### PR TITLE
TINY-9979: Implement schema presets

### PR DESCRIPTION
**Pull request by spocke**
Related Ticket: TINY-9979

Description of Changes:
* Added presets to `valid_elements`
* Refactored element sets so that global attributes are a separate thing.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts [hunk 1]
```diff
@@ -379,6 +379,61 @@ describe('browser.tinymce.core.html.SchemaTest', () => {
...
+    assert.isTrue(schema.isValidChild('strong', 'p'));
+
+    assert.isTrue(schema.isValidChild('em', 'span'));
+    assert.isFalse(schema.isValidChild('em', 'p'));
+    schema.addValidChildren('em[@phrasing|@blocks]');
+    assert.isTrue(schema.isValidChild('em', 'span'));
+    assert.isTrue(schema.isValidChild('em', 'p'));
+  });
+
+  it('TINY-9979: addValidChildren with a - prefixed preset should remoe preset', () => {
```
Comment (vpyshnenko) [inline modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts:410]:
```suggestion
  it('TINY-9979: addValidChildren with a - prefixed preset should remove preset', () => {
```

File: modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts [hunk 2]
```diff
@@ -379,6 +379,61 @@ describe('browser.tinymce.core.html.SchemaTest', () => {
...
+    const schema = Schema();
+
+    assert.isTrue(schema.isValidChild('div', 'span'));
+    assert.isTrue(schema.isValidChild('div', 'p'));
+    schema.addValidChildren('-div[@phrasing]');
+    assert.isFalse(schema.isValidChild('div', 'span'));
+    assert.isTrue(schema.isValidChild('div', 'p'));
+  });
+
+  it('TINY-9979: addValidChildren with a + prefixed preset should remoe preset', () => {
```
Comment (vpyshnenko) [inline modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts:420]:
```suggestion
  it('TINY-9979: addValidChildren with a + prefixed preset should append preset', () => {
```

File: modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts [hunk 3]
```diff
@@ -379,6 +379,61 @@ describe('browser.tinymce.core.html.SchemaTest', () => {
...
+    const schema = Schema();
+
+    assert.isTrue(schema.isValidChild('span', 'span'));
+    assert.isFalse(schema.isValidChild('span', 'p'));
+    schema.addValidChildren('+span[@blocks]');
+    assert.isTrue(schema.isValidChild('span', 'span'));
+    assert.isTrue(schema.isValidChild('span', 'p'));
+  });
+
+  it('TINY-9979: Add valid children though valid_children setting', () => {
```
Comment (vpyshnenko) [inline modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts:430]:
```suggestion
  it('TINY-9979: Add valid children through valid_children setting', () => {
```

**Review by vpyshnenko — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts (hunk 1 above)
Comment (spocke) [reply modules/tinymce/src/core/test/ts/browser/html/SchemaTest.ts]: Thanks for finding these typos. 👍

**Pull request merged**